File: src/main/java/com/example/ssauc/user/search/repository/SearchKeywordRepository.java
package com.example.ssauc.user.search.repository;


import com.example.ssauc.user.search.entity.SearchKeyword;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface SearchKeywordRepository extends JpaRepository<SearchKeyword, String> {
  Optional<SearchKeyword> findByKeyword(String keyword);
  List<SearchKeyword> findTop10ByOrderBySearchCountDesc();
}

File: src/main/java/com/example/ssauc/user/search/repository/SearchKeywordRepository.kt
package com.example.ssauc.user.search.repository

import com.example.ssauc.user.search.entity.SearchKeyword
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*


interface SearchKeywordRepository : JpaRepository<SearchKeyword?, String?> {
    fun findByKeyword(keyword: String?): Optional<SearchKeyword?>?
    fun findTop10ByOrderBySearchCountDesc(): List<SearchKeyword?>?
}